### PR TITLE
`ocaml-lsp` should be renamed to `ocaml-lsp-server`

### LINKED ISSUE #444: `ocaml-lsp` should be renamed to `ocaml-lsp-server`
```sh
omppye@macOS try-melange % opam install ocaml-lsp dot-merlin-reader
[ERROR] No package named ocaml-lsp found.
```

### PR DESCRIPTION
**Pull request by omppye**
Closes #444

### CODE REVIEW

File: nix/shell.nix [hunk 1]
```diff
@@ -39,7 +39,7 @@ mkShell {
     reason
     utop
     ocamlformat
-    ocaml-lsp
+    ocaml-lsp-server
```
Comment (omppye) [inline nix/shell.nix:42]: I don't know if this is a variable name

Comment (anmonteiro) [reply nix/shell.nix:42]: This was correct before. Nix has its own registry where it’s called ocaml-lsp

Comment (omppye) [reply nix/shell.nix:42]: I didn't check if in the Nix registry was called `ocaml-lsp-server` by the way. I was thinking that packages are from opam repository

Comment (anmonteiro) [reply nix/shell.nix:42]: It was correct before, please revert https://github.com/NixOS/nixpkgs/blob/master/pkgs/top-level/ocaml-packages.nix#L997

Comment (omppye) [reply nix/shell.nix:42]: 👍🏻

**Comment by anmonteiro** [comment]
Thanks

**Pull request merged**